test/integration/server-selection/server_selection.prose.operation_count.test.ts

```diff
@@ -45,13 +44,12 @@ describe('operationCount-based Selection Within Latency Window - Prose Test', fu
   let seeds: Array<string>;
   let counts: Record<string, number> = {};
   const updateCount = ({ address }: CommandStartedEvent) => {
-    const mongosPort = address.split(':')[1];
+    const portRegex = /(\d{5})/;
+    const [mongosPort] = portRegex.exec(address) ?? [''];
```

Comment:
Why exactly 5 digits, couldn't a port be any length? I assume this came from a test failure, what was the address that was coming through here? maybe we can use HostAddress helpers

Reply:
All our tests run with mongoses with 5 digit ports

Reply:
the address on Node18 is `::1:<port>`

Reply:
https://github.com/mongodb/node-mongodb-native/blob/8338bae933c777dee4e7e49dbcf52c4fd7047528/src/utils.ts#L1210

Reply:
Our tests accept a connection string from the env so the port could be any length, if someone used a 4 digit port the error from this test would be about the counts being wrong, so there wouldn't be an indication that a 5 digit port was a requirement